src/frontend/src/tests/btc/schema/btc-post-message.schema.spec.ts

```diff
@@ -0,0 +1,51 @@
+import { BtcPostMessageDataResponseWalletSchema } from '$btc/schema/btc-post-message.schema';
+import type { CertifiedData } from '$lib/types/store';
+
+describe('btc-post-message.schema', () => {
+	describe('BtcPostMessageDataResponseWalletSchema', () => {
+		const mockValidBalance: CertifiedData<bigint> = {
+			data: 1000n,
+			certified: true
+		};
+		const mockValidTransactions = JSON.stringify([{ id: 'tx1', amount: 500 }]);
+
+		it('should validate with a valid balance and newTransactions', () => {
+			const validData = {
+				wallet: {
+					balance: mockValidBalance,
+					newTransactions: mockValidTransactions
+				}
+			};
+			expect(BtcPostMessageDataResponseWalletSchema.parse(validData)).toEqual(validData);
+		});
+
+		it('should fail if newTransactions is missing', () => {
+			const invalidData = {
+				wallet: {
+					balance: mockValidBalance
+				}
+			};
+			expect(() => BtcPostMessageDataResponseWalletSchema.parse(invalidData)).toThrow();
+		});
+
+		it('should validate if balance is not a bigint because of zod custom', () => {
```

Comment:
that is why I am afraid of zod custom eheh! but nothing we can do i imagine

Reply:
We have some TODOs to remove the custom but yeah, as long we use custom, nothing we can do except writing a refine but, not sure it's worth it if we want to get rid of the custom in first place